src/coreclr/scripts/superpmi.py

```diff
@@ -2128,6 +2164,151 @@ def get_mch_files_for_replay(coreclr_args):
     return mch_files
 
 
+def process_base_jit_path_arg(coreclr_args):
+    """ Process the -base_jit_path argument coreclr_args.base_jit_path.
+        If the argument is present, check it for being a path to a file.
+        If not present, try to find and download a baseline JIT based on the current environment:
+        1. Determine the current git hash using:
+             git rev-parse HEAD
+           or use the `-git_hash` argument (call the result `git_hash`).
+        2. Determine the baseline: where does this hash meet `master` using:
+             git merge-base `git_hash` master
+           or use the `-base_git_hash` argument (call the result `base_git_hash`).
+        3. If the `-base_git_hash` argument is used, use that directly as the exact git
+           hash of the baseline JIT to use.
+        4. Otherwise, figure out the latest hash, starting with `base_git_hash`, that contains any changes to
+           the src\coreclr\src\jit directory. (We do this because the JIT rolling build only includes
+           builds for changes to this directory. So, this logic needs to stay in sync with the logic
+           that determines what causes the JIT directory to be rebuilt. E.g., it should also get
+           rebuilt if the JIT-EE interface GUID changes. Alternatively, we can take the entire list
+           of changes, and probe the rolling build drop for all of them.)
+        5. Check if we've already downloaded a JIT that matches `base_git_hash`, and use that if available.
+        6. Starting with `base_git_hash`, and possibly walking to older changes, look for matching builds
+           in the JIT rolling build drops.
+        7. If a baseline clrjit is found, download it to the `spmi/basejit/git-hash.os.architecture.build_type`
+           cache directory.
+        8. Set coreclr_args.base_jit_path to the full path to the downloaded baseline JIT.
+
+    Args:
+        coreclr_args (CoreclrArguments) : parsed args
+
+    Returns:
+        Nothing
+
+        coreclr_args.base_jit_path is set to the path to the JIT to use for the baseline JIT.
+    """
+
+    if coreclr_args.base_jit_path is not None:
+        if not os.path.isfile(coreclr_args.base_jit_path):
+            raise RuntimeError("Specified -base_jit_path does not point to a file")
+        return
+
+    # Used for debugging
+    verbose = False
+
+    # We cache baseline jits under the following directory. Note that we can't create the full directory path
+    # until we know the baseline JIT hash.
+    default_basejit_root_dir = os.path.join(coreclr_args.spmi_location, "basejit")
```

Comment:
I have multiple repos -- it would be nice if this cache could be machine wide. But you don't need to change that just now

Reply:
It would be easy to take an option to allow specifying where in the machine to place the cache. But that isn't great for usage simplicity. Maybe this script should use/share the `JIT_UTILS_ROOT` variable and config.json that the jitutils tools use to allow specifying defaults. (I wonder if anyone even knows this exists...)